controllers/etcdpeer_controller.go

```diff
@@ -162,6 +179,53 @@ func defineReplicaSet(peer etcdv1alpha1.EtcdPeer) appsv1.ReplicaSet {
 	}
 }
 
+func pvcForPeer(peer *etcdv1alpha1.EtcdPeer) *corev1.PersistentVolumeClaim {
+	return &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      peer.Name,
+			Namespace: peer.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(peer, etcdv1alpha1.GroupVersion.WithKind("EtcdPeer")),
+			},
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
```

Comment:
is there a reason this isn't set to `peer.Spec.VolumeClaimTemplate.Spec` directly? as a user, id be confused if i'd set some fields on the PVC and the actual behaviour ended up being different

Reply:
Yeah, this is why I think I made a mistake in using PVC directly. 
I don't want to support all its fields right now.

Reply:
why not? we can offer a supported workflow through the cluster controller (which can hold opinion on what the PVCs look like) while allowing people to make modifications to peers if they need

Reply:
I've updated it to DeepCopy the entire PVC spec, as you suggested.

Let's discuss it today.

I also added a defaulting function to ensure that all the fields required by PVC validation but which are optional fin our API are filled in.